### DISCUSSION TITLE
Forcing_variables.txt LIS input

### DISCUSSION THREAD

**Discussion by cbachand** [NASA-LIS/LISF#1070] [Q&A] [answered]
Hi, I'm wondering if there is any reason not to use all possible model forcings when running Noah-MP on LIS? I am working on two different model runs (one forced by MERRA2, one forced by ERA5) and am not exactly sure what forcing variables to use. 

Thank you!

**Comment by karsenau** [comment, marked as answer]
Hi @cbachand:

Noah-MP requires the following base forcing fields to run: 

Tair:       1  1  K       # Near surface air temperature
Qair:       1  1  kg/kg   # Near surface specific humidity
SWdown:     1  1  W/m2    # Incident shortwave radiation (total)
LWdown:     1  1  W/m2    # Incident longwave radiation
Wind_E:     1  1  m/s     # Eastward wind
Wind_N:     1  1  m/s     # Northward wind
Psurf:      1  1  Pa      # Surface pressure
Rainf:      1  1  kg/m2s  # Rainfall rate

Some forcing datasets do offer convective rainfall or possibly snowfall, so you can select these options in LIS, if they are available and if the LSMs can utilize them.

Please let us know if that answered your question.

**Reply by cbachand** [reply]
If I select a variable that isn't available in my forcing data set or can't be utilized by the LSM, will it cause an error? Or will LIS continue to run and just ignore that forcing?

**Reply by karsenau** [reply]
I don't believe it would cause an error if you run with a forcing field selected that is not in the forcing dataset. LIS should expect to continue to run and ignore those forcing fields that are not present.

**Reply by cbachand** [reply]
I've found that running Noah-MP with MERRA2 forcings, including  the snowfall forcing, results in much larger snow depth estimates than running Noah-MP with Merra2 forcings but without the snowfall forcing enabled. However, when I do the same with ERA5 forcings instead, the model estimates of snow depth are identical. Further, I outputted the snowfall forcing variable for the ERA5 run, and it was all 0s, even though there appears to be a Snowfall variable in the ERA5 forcings. 

Can you help me understand why I am observing this behavior? Why does enabling snowfall forcings make such a big difference for MERRA2, but no difference for ERA5? 

When I run Noah-MP without a snowfall forcing, does it partition rainfall into rainfall and snowfall depending on the air temp, etc.? Does this partitioning behavior stop when a snowfall forcing is added, or is possible that Noah-MP "double counts" snowfall when both a snowfall and rainfall forcing is provided? 

Thank you! I hope my questions make sense. 
Claire

**Reply by dmocko** [reply]
Hi @cbachand - yes, your questions make sense.

I will answer your later questions first.  Noah-MP-4.0.1 has four different physics options for partitioning precipitation into rainfall and snowfall.  See the LIS Users' Guide config entry - `Noah-MP.4.0.1 rainfall & snowfall option:`.  There are 3 options supported in LIS; one from Jordan (1991), one from the BATS LSM, and one from the Noah LSM.  They all use the air temperature in the partitioning.  Noah-MP LSM has one more option (to use the WRF microphysics partitioning), but it is not supported in LIS.

When a snowfall forcing is added to Noah-MP, the snowfall is added to the rainfall to get a total precipitation - which is then partitioned into rainfall/snowfall by one of the above options.  You can see where the snow forcing is added to get the total precip, starting on Line 187 in this routine: https://github.com/NASA-LIS/LISF/blob/master/lis/surfacemodels/land/noahmp.4.0.1/NoahMP401_f2t.F90#L187

Because most of the forcings in LIS only have total precip (and not a separate rainfall and snowfall forcing) - and because the ALMA conventions did not provide a total precip variable (only Rainf and Snowf) - AND because most LSMs only accept the total precip (and not rainfall and snowfall separately, such as Noah-MP) - the total precipitation amount is often put into the "Rainf" forcing variable.  But it appears that with the MERRA-2 forcing, the snowfall (Snowf) forcing represents the portion of the total precipitation (put in the Rainf variable).  So, when running with MERRA-2 forcing with snowfall forcing enabled, LIS will double-count the snow (when on Line 187 above, it adds the snowfall into the total precip).  That is why you are seeing larger snow depths with the snowfall forcing enabled for MERRA-2.

For MERRA-2, I recommend just turning off the Snowfall forcing, and letting Noah-MP do the partitioning with one of the above options.  Longer term, the LIS team should fix the MERRA-2 forcing code, so that when Snowfall forcing is turned on, that amount is subtracted from the "Rainf", so snowfall and rainfall are separated.

Earlier, I mentioned the WRF microphysics partitioning option, which is not supported in LIS.  From looking at the code, it seems possible to enable support for this option, so that Noah-MP accepts both rainfall and snowfall separately from the forcing.  Thus, the partitioning is done by the forcing, and not by the Noah-MP LSM.  We generally tend to let the Noah-MP LSM do the partitioning, as the air temperature (through interpolation and lapse-rate correction) may be a better indicator at that exact grid point, instead of the forcing, which tends to be at a coarser scale than the running LSM grid.

For ERA-5, looking at the code, it appears that the Rainf and Snowf fields are separately read from the forcing, then summed for a total precipitation.  The total precip is then put into the "Rainf" variable, and the "Snowf" field is not saved.  That is why you are seeing zeros for the Snowf from ERA-5, and enabling Snowf or not as a forcing variable does not make a difference in the output.  See this section of code: https://github.com/NASA-LIS/LISF/blob/master/lis/metforcing/era5/read_era5.F90#L247-L268

David
